src/sentry/coreapi.py

```diff
@@ -215,12 +182,51 @@ def insert_data_to_database(self, data, start_time=None,
                    event_id=data['event_id'])
 
 
-class MinidumpApiHelper(ClientApiHelper):
-    def origin_from_request(self, request):
+class ClientAuthHelper(object):
```

Comment:
Defining all of these as static methods, then subclassing this class, and overriding all of the methods without ever calling the superclass is a strange pattern, in my opinion.

Reply:
Would changing the subclasses to subclass `object` instead fix that, or is it something else about it?

I guess my thinking is this is still an organization of auth methods, and if someone wanted to add another method to `ClientAuthHelper` they should be able to safely because all `auth_helper_cls` references need to implement the same thing. But I'm not at all tied to that.

Reply:
If it was up to me, I'd probably use an ABCMeta base class and have all the subclasses extend that abstract interface instead since they don't share any functionality.

I don't think the class method thing is inherently an issue, but usually classes that are entirely static methods give me pause because I've often found them to be methods of working around other muddy abstractions.

That's obviously all subjective, it'll work just fine either way/as-is.

Reply:
I always forget about the `abc` module. This was a shitty attempt at an interface, so I'll switch to `abc`.

Agreed about entirely static, but I'm trying to avoid going refactor heavy vs just extracting what's already there (these effectively static methods that weren't marked as such) into a workable solution for the Kafka consumer. In theory all of this code is dead eventually as Relay will be publishing to Kafka itself.

Reply:
> Agreed about entirely static, but I'm trying to avoid going refactor heavy vs just extracting what's already there (these effectively static methods that weren't marked as such) into a workable solution for the Kafka consumer. In theory all of this code is dead eventually as Relay will be publishing to Kafka itself.

Yep, understood.